src/python_testing/TC_RR_1_1.py

```diff
@@ -0,0 +1,432 @@
+        self._subscriptions = []
+
+    def teardown_class(self):
+        logging.info("Teardown: shutting down all subscription to avoid racy callbacks")
+        for subscription in self._subscriptions:
+            subscription.Shutdown()
+
+    @async_test_body
+    async def test_TC_RR_1_1(self):
+        dev_ctrl = self.default_controller
+
+        # Debug/test arguments
+
+        # Get overrides for debugging the test
+        num_fabrics_to_commission = self.user_params.get("num_fabrics_to_commission", 5)
+        num_controllers_per_fabric = self.user_params.get("num_controllers_per_fabric", 3)
+        # Immediate reporting
+        min_report_interval_sec = self.user_params.get("min_report_interval_sec", 0)
+        # 10 minutes max reporting interval --> We don't care about keep-alives per-se and
+        # want to avoid resubscriptions
+        max_report_interval_sec = self.user_params.get("max_report_interval_sec", 10 * 60)
+        # Time to wait after changing NodeLabel for subscriptions to all hit. This is dependant
+        # on MRP params of subscriber and on actual min_report_interval.
+        # TODO: Determine the correct max value depending on target. Test plan doesn't say!
+        timeout_delay_sec = self.user_params.get("timeout_delay_sec", max_report_interval_sec * 2)
+        # Whether to skip filling the UserLabel clusters
+        skip_user_label_cluster_steps = self.user_params.get("skip_user_label_cluster_steps", False)
+
+        BEFORE_LABEL = "Before Subscriptions 12345678912"
+        AFTER_LABEL = "After Subscriptions 123456789123"
+
+        # Pre-conditions
+
+        # Make sure all certificates are installed with maximal size
+        dev_ctrl.fabricAdmin.certificateAuthority.maximizeCertChains = True
+
+        # TODO: Do from PICS list. The reflection approach here what a real client would do,
+        #       and it respects that the test says "TH writes 4 entries per endpoint where LabelList is supported"
+        logging.info("Pre-condition: determine whether any endpoints have UserLabel cluster (ULABEL.S.A0000(LabelList))")
+        endpoints_with_user_label_list = await dev_ctrl.ReadAttribute(self.dut_node_id, [Clusters.UserLabel.Attributes.LabelList])
+        has_user_labels = len(endpoints_with_user_label_list) > 0
+        if has_user_labels:
+            logging.info("--> User label cluster present on endpoints %s" %
+                         ", ".join(["%d" % ep for ep in endpoints_with_user_label_list.keys()]))
+        else:
+            logging.info("--> User label cluster not present on any endpoitns")
+
+        # Generate list of all clients names
+        all_names = []
+        for fabric_idx in range(num_fabrics_to_commission):
+            for controller_idx in range(num_controllers_per_fabric):
+                all_names.append("RD%d%s" % (fabric_idx + 1, chr(ord('A') + controller_idx)))
+        logging.info("Client names that will be used: %s" % all_names)
```

Comment:
FYI in general, you can just in-band expand the variables instead of using format specifiers, i.e `f"Client names that will be used {all_names}"` instead

Reply:
I am just much more used to the legacy way, Will change a few